File: fix2.py
path = r'C:\caas-dashboard\frontend\src\MigrationPage.jsx'
raw = open(path, 'rb').read()

old = b'cancelPlan(plan.id)'
new = b'doUpdateStatus(plan.id, "cancelled")'

# Only replace first occurrence (the table row one at line 1250)
# Check it's the right one
lines = raw.split(b'\r\n')
count = raw.count(old)
print(f'Found {count} occurrence(s) of cancelPlan(plan.id)')
if count == 1:
    raw = raw.replace(old, new, 1)
    open(path, 'wb').write(raw)
    print('Done - replaced with doUpdateStatus')
else:
    print('Multiple or zero - manual fix needed')

File: fix2.js
const fs=require("fs");
const f="C:\\caas-dashboard\\frontend\\src\\App.jsx";
const lines=fs.readFileSync(f,"utf8").split("\n");
let fixed=0;
for(let i=0;i<lines.length;i++){
  const noToken=!lines[i].includes("api_token");
  const hasPass=lines[i].includes("port:form.port.trim(), username:form.user.trim(), password:form.pass.trim()");
  const nextClose=lines[i+1]&&lines[i+1].trim()==="});";
  if(hasPass&&noToken&&nextClose){
    lines[i]=lines[i].replace("password:form.pass.trim()","password:form.pass.trim(), api_token:(form.api_token||\"\").trim()");
    console.log("Fixed testStorageConnection at line "+(i+1));
    fixed++;
  }
}
if(!fixed){console.error("Not found");process.exit(1);}
fs.writeFileSync(f,lines.join("\n"),"utf8");
console.log("Saved "+fixed+" fix(es)");
